Riot/Modules/MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -2185,10 +2185,32 @@ - (void)resendEventWithEventId:(NSString *)eventId success:(void (^)(NSString *)
             [self removeEventWithEventId:eventId];
             
             if (event.isVoiceMessage) {
-                NSNumber *duration = event.content[kMXMessageContentKeyExtensibleAudioMSC1767][kMXMessageContentKeyExtensibleAudioDuration];
-                NSArray<NSNumber *> *samples = event.content[kMXMessageContentKeyExtensibleAudioMSC1767][kMXMessageContentKeyExtensibleAudioWaveform];
-                
-                [self sendVoiceMessage:localFileURL mimeType:mimetype duration:duration.doubleValue samples:samples success:success failure:failure];
+                // Check if it is an actual voice message or a voicebroadcast chunk
+                if (event.content[VoiceBroadcastSettings.voiceBroadcastContentKeyChunkType] != nil) {
```

Comment:
Instead of switching on the type I would just build the `additionalContentParams` conditionally and then use the same `sendVoiceMessage` method for both with an empty dictionary for voice messages